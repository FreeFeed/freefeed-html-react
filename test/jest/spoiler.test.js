/* global describe, it, expect */
import { render, screen, fireEvent } from '@testing-library/react';

import Spoiler from '../../src/components/spoiler';

describe('Spoiler', () => {
  it('Renders correctly', () => {
    const { asFragment } = render(<Spoiler>Aeris dies</Spoiler>);

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(screen.getByTitle('This is a spoiler. Click to reveal its contents.'));
    expect(asFragment()).toMatchSnapshot();
  });
});
