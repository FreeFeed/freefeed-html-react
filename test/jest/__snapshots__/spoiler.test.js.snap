// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spoiler Renders correctly 1`] = `
<DocumentFragment>
  <span
    class="spoiler-hidden"
    title="This is a spoiler. Click to reveal its contents."
  >
    Aeris dies
  </span>
</DocumentFragment>
`;

exports[`Spoiler Renders correctly 2`] = `
<DocumentFragment>
  <span
    class="spoiler-visible"
  >
    Aeris dies
  </span>
</DocumentFragment>
`;
